### PR TITLE
feat: add csv parsing for meshes and tutorial

### PR DESCRIPTION
**Pull request by guenthermi**
Add csv parsing for meshes and tutorial 

---

- [x] This PR references an open issue https://github.com/jina-ai/finetuner-core/issues/420
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: docs/notebooks/mesh_to_mesh.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+```
+
+<!-- #region id="pfoc4YG4rrkI" -->
+And finally you can use the embeded `query` to find top-k visually related images within `index_data` as follows:
+<!-- #endregion -->
+
+```python id="_jGsSyedrsJp"
+query.match(index_data, limit=10, metric='cosine')
+```
+
```
Comment (bwanglzu) [inline docs/notebooks/mesh_to_mesh.md:222]: let's add a new section called `before and after`

File: docs/notebooks/mesh_to_mesh.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+<!-- #region id="JU3uUVyirTE1" -->
+## Inference
+
+Now you saved the `artifact` into your host machine,
+let's use the fine-tuned model to encode a new `Document`:
+
+```{admonition} Inference with ONNX
+In case you set `to_onnx=True` when calling `finetuner.fit` function,
+please use `model = finetuner.get_model(artifact, is_onnx=True)`
```
Comment (bwanglzu) [inline docs/notebooks/mesh_to_mesh.md:200]: is it True for pointnet? have we resolved the issue of onnx conversion?

Comment (bwanglzu) [reply docs/notebooks/mesh_to_mesh.md:200]: if not we need to add a warning message in the `fit` section

**Review by bwanglzu — CHANGES_REQUESTED** [review]
please also fix the `describe_models`
D0AE6CA2-9D34-40AA-A69B-EBB3F7C2C28A

File: finetuner/data.py [hunk 3]
```diff
@@ -32,6 +32,10 @@ class CSVOptions:
         should be assigned to the item in the first column (True), or if it is another
         item that should be semantically close to the first (False).
     :param convert_to_blob: Whether uris to local files should be converted to blobs
+    :param create_point_clouds: Determines the number of points sampled from a mesh to
+        create a point cloud.
+    :param point_cloud_size: Determines the number of points sampled from a mesh to
+        create a point cloud.
```
Comment (LMMilliken) [inline finetuner/data.py:38]: The description for `create_point_clouds` is wrong

File: finetuner/data.py [hunk 4]
```diff
@@ -211,6 +242,13 @@ def create_document(
                 doc.load_uri_to_blob()
         else:
             raise ValueError(f'Expected uri in column 1, got {column}')
+    elif modality == 'mesh':
+        if _is_uri(column):
+            doc = Document(uri=column)
+            if create_point_clouds and os.path.isfile(column):
+                doc.load_uri_to_point_cloud_tensor(point_cloud_size)
```
Comment (LMMilliken) [inline finetuner/data.py:249]: this document would be unusable if `create_point_clouds` is False but `column` is a local file right?

**Review by LMMilliken — COMMENTED** [review]
two small things

File: finetuner/data.py (hunk 4 above)
Comment (guenthermi) [reply finetuner/data.py:249]: yes more or less, but it is the same for images when you set `convert_to_blob=False`.

Comment (LMMilliken) [reply finetuner/data.py:249]: good point, maybe both of these cases should raise an error or warning

File: docs/notebooks/mesh_to_mesh.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+Finding similar 3D Meshes can become very time-consuming. To support this task, one can build search systems. To directly search on the 3D meshes without relying on metadata one can use an encoder model which creates a point cloud from the mesh and encode it into vector dense representations which can be compared to each other. To enable those models to detect the right attributes of a 3D mesh, this tutorial shows you how to use Finetuner to train and use a model for a 3D mesh search system.
+<!-- #endregion -->
+
+<!-- #region id="mk4gxLZnYJry" -->
+## Install
+<!-- #endregion -->
+
+```python id="vDVkw65kkQcn"
+!pip install 'finetuner[full]'
+!pip install 'docarray[full]'
```
Comment (bwanglzu) [inline docs/notebooks/mesh_to_mesh.md:28]: let's remove this from the tutorial

**Review by bwanglzu — APPROVED** [review]
LGTM!, minor comment

**Comment by guenthermi** [comment]
> let's remove this from the tutorial

Oh yes, this is terrible. I can add a white backround surronding all the image of the meshes

File: CHANGELOG.md [hunk 6]
```diff
@@ -14,6 +14,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Add `core-ci` workflow to remotely run the ci of finetuner-core. ([#628](https://github.com/jina-ai/finetuner/pull/628))
 
+- Ass support for 3d meshes to `build_finetuning_dataset`. ([#638](https://github.com/jina-ai/finetuner/pull/638))
+
```
Comment (LMMilliken) [inline CHANGELOG.md:18]: typo

**Review by LMMilliken — CHANGES_REQUESTED** [review]
One small thing

**Review by LMMilliken — APPROVED** [review]
LGTM

**Comment by bwanglzu** [comment]
is there an error in `hit@k` metric? seems it's not improving
